yt/frontends/flash/data_structures.py

```diff
@@ -99,6 +99,9 @@ def _parse_index(self):
         self.grid_dimensions[:] *= (nxb, nyb, nzb)
         try:
             self.grid_particle_count[:] = f_part["/localnp"][:][:, None]
+            self._blockless_particle_count = (
+                f_part["/tracer particles"].shape[0] - self.grid_particle_count.sum()
+            )
         except KeyError:
             self.grid_particle_count[:] = 0.0
```

Comment:
I'm going to decline the suggestion, because this makes it considerable longer, and the original logic explicitly ties the two attributes together, which they should be.  Unless @jzuhone disagrees, I'd like this not to be changed.

Reply:
Although I will note I *will* definitely add the `blockless_particle_count` initializer -- thanks for that catch!  What I'm declining is splitting the try blocks.

Reply:
If they're tied up _by design_ it suits me just fine :)